packages/app/src/cli/constants.ts

```diff
@@ -158,7 +160,15 @@ export function getUIExtensionRendererDependency(extensionType: UIExtensionTypes
 }
 
 export const uiExternalExtensionTypes = {
-  types: ['web_pixel', 'post_purchase_ui', 'checkout_ui', 'pos_ui', 'subscription_ui', 'customer_accounts_ui'],
+  types: [
+    'web_pixel',
+    'post_purchase_ui',
+    'checkout_ui',
+    'checkout_ui_beta',
```

Comment:
Should this be removed?

Reply:
`checkout_ui_beta`? No I don't think we can remove this, since it's used in a few places in this file: https://github.com/Shopify/cli/blob/add-new-checkout-ui-template/packages/app/src/cli/utilities/extensions/name-mapper.ts

That assume the approach I've taken in this PR is generally sound though:

* `Checkout UI (Beta)` for the human friendly name.
* `checkout_ui_beta` to map the human friendly name to a template in `packages/app/templates/ui-extensions/projects/checkout_ui_beta`
* `ui_extension` for the extension specification type (used to default the type value for the toml schema).

Reply:
I'm still confused as to why this needs to be exposed in `uiExternalExtensionTypes`. We only need to care about what template they want to use during the generate step. Can we just simplify for now by renaming `checkout_ui_beta` to `ui_extension`? Once the plugins system is built we can improve this to display an additional prompt for selecting template instead of hardcoding it to the checkout template.

Reply:
Discussed offline - we can leave this for now since we made the effort to get this to work and no good reason to undo it as the underlying type is still `ui_extension`